### PR TITLE
modify(android/app): Change build-help.sh to generate offline help from Markdown files

### PR DESCRIPTION
**Pull request by darcywong00**
Updates build-help.sh in addressing keymanapp/help.keyman.com#288

This PR updates the build-help.sh script for generating offline help. 
(Inspired by windows/src/desktop/help/build.sh usage of `pandoc` to convert Markdown to html files)

keymanapp/help.keyman.com#306 reorganized the Keyman for Android help from php to Markdown files

## Summary of commits
14e5e981feab0540ef0cb97070e35556b9878fbe - throwaway code, editing the old build-help.sh file (gets removed by the next commit)
45801f41a63cd5f3bd6afaadab15f160223e0d9d - replaces build-help.sh to use pandoc
644c2e2fc4651bdc3d1e259c23f29977b68447db - cherrypick the Markdown files from keymanapp/help.keyman.com#306
01310e3426c795cc18a6119eddaf5fa97f9c41e9 - auto-generated offline help from running
```bash
cd help
build-help.sh htm
```

I plan to implement the Android version of `windows/src/buildtools/help-keyman-com.sh` on a separate PR

## Screenshots

Network accessing help.keyman-staging.com/products/android/14.0
online

Note: Offline help won't have css styling
Offline help (index.html with new "About" section)
offline index

Offline menu buttons (phone layout)
offline menu-button

### CODE REVIEW

**Review by mcdurdin — CHANGES_REQUESTED** [review]
As the .html files are generated from the .md, they should not be included in the repo.

**Comment by darcywong00** [comment]
> As the .html files are generated from the .md, they should not be included in the repo.

But the .html files are needed in the assets/ folder so they can be included in the app as offline help.

The removed .php.html files on this PR were generated by the previous `build-help.sh`.

**Comment by mcdurdin** [comment]
The .html files should be generated as part of the build.

File: android/.gitignore [hunk 1]
```diff
@@ -41,6 +41,9 @@ KMEA/**/assets/resources
 # keymanweb OSK font
 KMEA/**/assets/keymanweb-osk.*
 
+# KMAPRo generated offline help
+KMAPro/**/info
```
Comment (mcdurdin) [inline android/.gitignore:45]: Is this just a single folder or two? If so, I think I'd prefer to have the full path for each one. Probably ditto on the other entries!

File: android/help/build-help.sh [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+#
+# Parse args
+#
+
+shopt -s nocasematch
+
+while [[ $# -gt 0 ]] ; do
+  key="$1"
+  case $key in
+    htm)
```
Comment (mcdurdin) [inline android/help/build-help.sh:45]: Why `htm` and not `html`? Given the generated files have the .html extension?

**Review by mcdurdin — APPROVED** [review]
LGTM. I haven't read through the documentation though ;-)

File: android/.gitignore (hunk 1 above)
Comment (darcywong00) [reply android/.gitignore:45]: It's a lot longer than two folders
`KMAPro\kMAPro\src\main\assets\info`

Maybe a middle ground of `KMAPro/**/assets/info`?

File: android/help/build-help.sh (hunk 2 above)
Comment (darcywong00) [reply android/help/build-help.sh:45]: I was replacing `chm` to `htm` from the original script. I can tweak to accept both `htm` and `html` :)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.235-beta](https://keyman.com/downloads/releases/beta/14.0.235)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.15-alpha](https://keyman.com/downloads/releases/alpha/15.0.15)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.16-alpha](https://keyman.com/downloads/releases/alpha/15.0.16)

**Pull request merged**